docs/installation/server-config/security.rst

```diff
@@ -0,0 +1,163 @@
+use any consumer group.
+
+The required ACLs are:
+
+- *DESCRIBE_CONFIGS* permission on the *CLUSTER*.
+- *CREATE* permission on the *CLUSTER*.
+- DESCRIBE*, *READ*, *WRITE* and *DELETE* permissions on the *<any>* *TOPIC*.
+- *DESCRIBE* and *READ* permissions  on the *<any>* *GROUP*.
+
+It is still possible to restrict the KSQL user from accessing specific resources using *DENY* ACLs. For example, you can add a
+*DENY* ACL to stop KSQL queries from accessing a topic that contains sensitive data.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Non-Interactive KSQL clusters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Non-interactive KSQL clusters will run with much more restrictive ACLs, though it currently requires a little effort to
+work out what ACLs are required. This will be improved in future KSQL releases.
+
+Standard ACLs
+    The KSQL user always requires:
+
+    - *DESCRIBE_CONFIGS* permission on the *CLUSTER*.
+    - *DESCRIBE* permission on the *__consumer_offsets* topic.
+
+    If you want KSQL to create internal and sink topics then the KSQL user should also be granted:
+
+    - *CREATE* permission on the *CLUSTER*.
+
+Source topics
+    KSQL users require *DESCRIBE* and *READ* permissions for each source and input topic. The topic should already exist
+    when KSQL is started.
+
+Sink topics
+    KSQL users require *DESCRIBE* and *WRITE* permissions ror each sink and output topic. If the topic does not already
+    exist, the user also requires *CREATE* permissions on the *CLUSTER*.
+
+Change-log and repartition topics
+    The set of changelog and repartitioning topics that KSQL requires depends on the queries being executed. The easiest
+    way to determine the list of required topics is to first run the queries on an open Kafka cluster and list the topics
+    that are created.
+
+    All changelog and repartition topics are prefixed with  ``<value of ksql.service.id property>_query_<query id>_`` where
+    the default of ``ksql.service.id`` is ``ksql_``.
+
+    KSQL users require a minimum of *DESCRIBE*, *READ* and *WRITE* permissions for each changelog and repartition *TOPIC*.
+
+    If the KSQL user does not have *CREATE* permissions on the *CLUSTER*, then all changelog and repartition topics must
+    already exist, with the same number of partitions as the source topic, and ``replication.factor`` replicas.
+
+Consumer groups
+    The set of consumer groups that KSQL requires depends on the queries that are being executed. The easiest way to
+    determine the list of consumer groups is to first run the queries on an open Kafka cluster and list the groups created.
+
+    Consumer group names are formatted like ``<value of ksql.service.id property>_query_<query id>``, where the default
+    of ``ksql.service.id`` is ``ksql_``.
+
+    KSQL users require a minimum of *DESCRIBE* and *READ* permissions for *GROUP*.
+
+-----------------------------
+Configuring Security for KSQL
```

Comment:
We have "Securing KSQL" above, so what is this about? (Yes, I can read the content, but let's clarify the headings.)

Reply:
Also, this section looks like braindump. Can we structure this in a better way?

- For example, at the top of the file we have a bullet entry on KSQL with (secure) Confluent Schema Registry. Why is there no separate section for this then?
- I'd suggest to add an explicit "Secure KSQL configuration example" section, where we show the `ksql-server.properties` below, and also mention exactly *what* that example demonstrates (and what it is not).

Reply:
@miguno, if you take a look at the [FAQ](https://github.com/confluentinc/ksql/blob/4.1.x/docs/faq.rst#can-ksql-connect-to-an-apache-kafka-cluster-over-ssl), you'll see the config broken out into separate sections.

Reply:
nvm @big-andy-coates  -- @joel-hamill has already been revamping the target TOC in a separate PR.  This PR here will be folded into that.